cmd/experimental/kjobctl/pkg/builder/slurm_builder.go

```diff
@@ -446,7 +447,32 @@ func (b *slurmBuilder) build(ctx context.Context) (runtime.Object, []runtime.Obj
 		b.cpusPerGpu = resource.NewQuantity(cpusPerGpu, resource.DecimalSI)
 	}
 
-	return job, []runtime.Object{configMap, service}, nil
+	role := &rbacv1.Role{
+		TypeMeta:   metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: b.objectName, Namespace: b.namespace},
+		Rules: []rbacv1.PolicyRule{{
+			Verbs:     []string{"get", "list"},
+			APIGroups: []string{""},
+			Resources: []string{"pods"},
+		}},
+	}
+
+	roleBinding := &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: b.objectName, Namespace: b.namespace},
+		Subjects: []rbacv1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      "default",
+			Namespace: b.namespace,
+		}},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     b.objectName,
+		},
+	}
+
+	return job, []runtime.Object{configMap, service, role, roleBinding}, nil
```

Comment:
I don't think we should expect a `kjobctl` user to have the necessary permissions to create roles. 

Maybe we should have an admin focused command that will create the SA , role and binding. Then have the SA name specified somewhere in the slurm profile.

In here, creation of the slurm job, we can check if the SA exists, if not we can jus warn the user and skip the attempt to get the IP.

Reply:
@trasc @mimowo So can we skip it for current PR? WDYT?

Reply:
I mean `role` and `roleBinding` creation for admin.

Reply:
At least, an admin can create them during the creation of an ApplicationProfile. We just need to add some documentation on how to allow users to obtain an IP. And also we can add it on the samples.

Reply:
> install the role and roleBinding during installation of kjobctl

We don't know exactly which namespace user will use.

Reply:
The admin is also responsible for creating ApplicationProfile and VolumeBundle resources. Therefore, we can shift the responsibility to the admin for creating the role and RoleBinding for the current namespace.

Reply:
Ok, all the complications are needed only for SLURM_JOB_FIRST_NODE_IP, so please hide it behind the opt-in option, and we can describe in the docs all the necessary steps to use it.